File: app/src/main/java/io/github/garrettsummerfi3ld/callout/AlertSmsFragment.java
package io.github.garrettsummerfi3ld.callout;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

public class AlertSmsFragment extends DialogFragment {
    public Dialog onCreateDialog(Bundle savedInstanceState, int alertFlag) {
        // Use the Builder class for convenient dialog construction
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.app_name).setMessage(R.string.sms_perm_dialog)
                .setPositiveButton(R.string.ok_dialog, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        // User accepted the dialog
                    }
                })
                .setNegativeButton(R.string.cancel_dialog, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        // User cancelled the dialog
                    }
                })
                .setIcon(R.drawable.ic_priority_high_black_24dp);
        // Create the AlertDialog object and return it
        return builder.create();
    }
}

File: app/src/main/java/io/github/garrettsummerfi3ld/callout/AlertSmsFragment.kt
package io.github.garrettsummerfi3ld.callout

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog

class AlertSmsFragment : DialogFragment() {
    fun onCreateDialog(savedInstanceState: Bundle, alertFlag: Int): Dialog {
        // Use the Builder class for convenient dialog construction
        val builder = AlertDialog.Builder(activity!!)
        builder.setTitle(R.string.app_name).setMessage(R.string.sms_perm_dialog)
                .setPositiveButton(R.string.ok_dialog) { dialog, id ->
                    // User accepted the dialog
                }
                .setNegativeButton(R.string.cancel_dialog) { dialog, id ->
                    // User cancelled the dialog
                }
                .setIcon(R.drawable.ic_priority_high_black_24dp)
        // Create the AlertDialog object and return it
        return builder.create()
    }
}